### DISCUSSION TITLE
Anyway to change the SSO UX so that users don't have to click the SSO button when only one SSO provider is configured?

### DISCUSSION THREAD

**Discussion by danijam** [argoproj/argo-cd#13568] [Q&A]
As title. Just configured SSO with Azure OIDC. I expected when I navigated to Argo I would be directed to Azure IdP automatically but instead I land on the login page giving me the option to login using a local credential or click the Azure SSO button. From a UX perspective I don't want to be fielding questions from internal users who are entering their AAD credentials into the login form and wondering why they aren't being logged in... Is there anyway to configure this behaviour?

**Comment by claire-fletcher** [comment]
On the newest version of ArgoCD, unsure about older versions, when I removed all local users and disabled the local admin account (i.e. the default included account) this removed the login and left just my SSO login button. Is that the sort of UX you wanted? If so it might be worth a look!

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cm
data:
  admin.enabled: "false"
```

**Reply by danijam** [reply]
It was more about achieving a UX where ArgoCD automatically redirects an unauthenticated user to the IdP without requiring the user to click a button

**Reply by claire-fletcher** [reply]
Ah I see! Sorry, I'm not sure if that is possible at the moment. I would also be interested to know if that is possible though. We wanted something similar where the service is behind IaP and once the user is logged in with that they don't need to click anymore login buttons. I've been tracking this: https://github.com/argoproj/argo-cd/issues/8027 for that purpose.
